### DISCUSSION TITLE
Default permissions despite configuring sync_dir_permissions and sync_file_permissions

### DISCUSSION THREAD

**Discussion by TwentySeven28** [abraunegg/onedrive#2320] [Q&A] [answered]
Thank you for an awesome app!

When files are synced to Unraid they're created with default permissions (600, 700). I created the config file, made sure to remove the "#", updated sync_dir_permissions and sync_file_permissions both to "777", and pointed the conf container path at the correct location.

* Unraid latest build (6.11.5)
* onedrive latest build 2.4.23

Anyone have any ideas? Is there a log file of some kind I could upload?

**Comment by abraunegg** [comment]
@Gareth4321 
> When files are synced to Unraid they're created with default permissions (600, 700).

Correct. This is the *correct* security mode for your data. This is not a bug or issue.

> I created the config file, made sure to remove the "#", updated sync_dir_permissions and sync_file_permissions both to "777", and pointed the conf container path at the correct location.

Using a mode of 777 is essentially opening up a *major* access and security issue for you - you should not do this - ever. Do not use a mode of 777 - yes you *can* use it - but in reality - you should never do that ....

Why? The permission 777 means that any user on your operating system can modify, execute, and write to the files posing a significant security risk to your system. An unauthorised user could use this to modify files to compromise your system.

> Anyone have any ideas? 

You really should *fix* your Unraid security access model - fix your users / groups and assign these correctly across your platform and data. Please ask on the Unraid forums how to fix your user & group security as this is out of scope for here.

If you are *really* fixated on using 777 to fix your security access issue:
* As per the [Docker Advanced Setup](https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#advanced-setup)  - create a configuration volume, then in the 'onedrive_conf' volume edit your custom configuration file so that 'sync_dir_permissions' and 'sync_file_permissions' equal your desired state. It sounds like you have done this step.
* Once configured and saved - restart your Docker container so that it now picks up your desired configuration. It is unclear if you did this step.

> Is there a log file of some kind I could upload?

You can increase the logging level output by using `ONEDRIVE_VERBOSE=1` when starting your Docker container, for example:
```
docker container run -e ONEDRIVE_VERBOSE=1 -v onedrive_conf:/onedrive/conf -v "${ONEDRIVE_DATA_DIR}:/onedrive/data" driveone/onedrive:latest
```

You can also 'login' to your Docker container and run `onedrive --display-config` to show you the active running configuration that it is using.

**Reply by TwentySeven28** [reply]
Thank you very much for the helpful response!

You are absolutely right about the permissions. I am currently setting up a new server and if everything works as intended I will nail down the permissions.

I logged into the the container and the config is still using default permissions. I did find this interesting:

```
Config path = /root/.config/onedrive
Config file found in config path = false
```

I ran the following command to give me the bind location for the onedrive_conf volume:

```
root@Server:~# docker volume inspect onedrive_conf
[
    {
        "CreatedAt": "2023-02-26T21:54:43+01:00",
        "Driver": "local",
        "Labels": {},
        "Mountpoint": "/var/lib/docker/volumes/onedrive_conf/_data",
        "Name": "onedrive_conf",
        "Options": {},
        "Scope": "local"
    }
]
```

I am undoubtedly making a silly mistake here, but I tried placing the config file (which I named "onedrive_conf") in the local "/var/lib/docker/volumes/onedrive_conf/_data" directory. This did not work. Should I somehow bind that mountpoint to a local directory to ensure the onedrive_conf volume is pointing to the onedrive_conf file? Or should I try to copy onedrive_conf to the volume directly?

**Comment by abraunegg** [comment]
@Gareth4321 
> I am undoubtedly making a silly mistake here, but I tried placing the config file (which I named "onedrive_conf") in the local "/var/lib/docker/volumes/onedrive_conf/_data" directory.

This is the wrong process. Please re-read https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md especially step 2.

Your 'config' file *must* reside in the configuration volume. This is why in step 6 you *inspect* the volume and create your custom configuration there.

**Reply by TwentySeven28** [reply]
Thank you for additional input.

The volume has already been created as per step 2:

`docker volume create onedrive_conf
`

I then inspected the volume as per step 6:

`docker volume inspect onedrive_conf
`

This resulted in the following output:

```
[
    {
        "CreatedAt": "2023-02-26T21:54:43+01:00",
        "Driver": "local",
        "Labels": {},
        "Mountpoint": "/var/lib/docker/volumes/onedrive_conf/_data",
        "Name": "onedrive_conf",
        "Options": {},
        "Scope": "local"
    }
]
```
Now I am stuck. You say that I should _not_ place the config file in "/var/lib/docker/volumes/onedrive_conf/_data". Where should I put it?

**Comment by abraunegg** [comment]
@Gareth4321 
> Now I am stuck. You say that I should not place the config file in "/var/lib/docker/volumes/onedrive_conf/_data". Where should I put it?

You need to place a file called 'config' in your `/var/lib/docker/volumes/onedrive_conf/_data` directory as per the documentation and instructions.

**Reply by TwentySeven28** [reply]
The file is there.

* I have tried renaming the file from "onedrive_conf" to "config".
* I have tried removing the volume and creating it again.
* I have tried restarting and deleting the container and installing again.

Could it be a permission problem?

`-rw-rw-rw- 1 root root 1757 Feb 27 20:58 config`

**Comment by abraunegg** [comment]
@Gareth4321 
The existing Docker instructions are really clear and over 2M folk running the Docker container are not having issues like this - so this is 100% something with what you are doing or your setup or Unraid.

Your issue could also be compounded by your desire to use 777 as your file access mode. You have stated that this is a new server build - my suggestion is that you do things right from the start including user & group permissions and their assignment and use across the system you are setting up.

Suggest that you seek support from Unraid to assist you further.

**Comment by TwentySeven28** [comment]
You are most likely correct in that it is something I am doing wrong. I will persevere. Thank you for your assistance all the same.

**Comment by TwentySeven28** [comment, marked as answer]
I found a solution. I'm not sure which one of these worked but I can't spend more time troubleshooting:

1. I set permission on "/var/lib/docker/volumes/onedrive_conf/_data" and the config file to 777.

2. I named the configuration file "config" and placed it in the folder mapped to container path "/onedrive/conf/".

Note that logging into the container and checking the current config in use (using the commands below) doesn't appear to work for me. The config displays default permissions, but I have confirmed that the new config is currently in use.

```
docker exec -it <container ID> /bin/bash
onedrive --display-config
```

**Reply by abraunegg** [reply]
@Gareth4321 
> I set permission on "/var/lib/docker/volumes/onedrive_conf/_data" and the config file to 777

Stop using 777 and please learn how to configure Linux users and groups & permission bits correctly.

**Reply by TwentySeven28** [reply]
The way unraid works isn’t like how standard Linux systems work. They only run with the root and nobody users. As unraid is intended for home use, the devs at limetech have been trying to give people as little reason to manipulate files through the CLI.

Generally most dockers tend to run an unmask of 755 and that’s not needed to be messed with. Some set files to read only for config. This is mainly the linuxserver packages. It's common to advise people with issues to set permissions to 777 for troubleshooting.

In cyber security there is the concept that good enough is much better than perfect. It's much better to focus on simple, effective, and achievable safety for home users than strict enforcement of, for example, permissions. In the latter, compliance is poor and users just stop caring about security at all.

Anyway, thanks again for your time and work on this application.
